applications/teletype/src/templates/teletype_port_cancel.erl

```diff
@@ -0,0 +1,193 @@
+fix_numbers(JObj) ->
+    Numbers =
+        wh_json:foldl(
+            fun(Number, _Value, Acc) ->
+                [Number|Acc]
+            end
+            ,[]
+            ,wh_json:get_value(<<"numbers">>, JObj)
+        ),
+    wh_json:set_value(<<"numbers">>, Numbers, JObj).
+
+-spec fix_billing(wh_json:object()) -> wh_json:object().
+fix_billing(JObj) ->
+    wh_json:foldl(
+        fun(Key, Value, Acc) ->
+            wh_json:set_value(<<"bill_", Key/binary>>, Value, Acc)
+        end
+        ,wh_json:delete_key(<<"bill">>, JObj)
+        ,wh_json:get_value(<<"bill">>, JObj)
+    ).
+
+
+-spec handle_port_request(wh_json:object()) -> 'ok'.
+-spec handle_port_request(wh_json:object(), wh_proplist()) -> 'ok'.
+handle_port_request(DataJObj) ->
+    teletype_util:send_update(DataJObj, <<"pending">>),
+    handle_port_request(DataJObj, teletype_util:fetch_templates(?TEMPLATE_ID, DataJObj)).
+
+handle_port_request(_DataJObj, []) ->
+    lager:debug("no templates to render for ~s", [?TEMPLATE_ID]);
+handle_port_request(DataJObj, Templates) ->
+    ServiceData = teletype_util:service_params(DataJObj, ?MOD_CONFIG_CAT),
+
+    Macros = [{<<"service">>, ServiceData}
+              ,{<<"account">>, teletype_util:public_proplist(<<"account">>, DataJObj)}
+              ,{<<"port_request">>, teletype_util:public_proplist(<<"port_request">>, DataJObj)}
+             ],
+
+    RenderedTemplates = [{ContentType, teletype_util:render(?TEMPLATE_ID, Template, Macros)}
+                         || {ContentType, Template} <- Templates
+                        ],
+
+    {'ok', TemplateMetaJObj} =
+        teletype_util:fetch_template_meta(?TEMPLATE_ID
+                                          ,teletype_util:find_account_id(DataJObj)
+                                         ),
+
+    Subject =
+        teletype_util:render_subject(
+            wh_json:find(<<"subject">>, [DataJObj, TemplateMetaJObj], ?TEMPLATE_SUBJECT)
+            ,Macros
+        ),
+
+    Emails = teletype_util:find_addresses(DataJObj, TemplateMetaJObj, ?MOD_CONFIG_CAT),
+
+    case
+        teletype_util:send_email(
+            Emails
+            ,Subject
+            ,ServiceData
+            ,RenderedTemplates
+            ,get_attachments(DataJObj)
+        )
+    of
+        'ok' ->
+            teletype_util:send_update(DataJObj, <<"completed">>);
+        {'error', Reason} ->
+            teletype_util:send_update(DataJObj, <<"failed">>, Reason)
+    end.
+
+-spec get_attachments(wh_json:object()) -> attachments().
+-spec get_attachments(wh_json:object(), boolean()) -> attachments().
+get_attachments(DataJObj) ->
+    get_attachments(DataJObj, teletype_util:is_preview(DataJObj)).
+
+get_attachments(_DataJObj, 'true') -> [];
+get_attachments(DataJObj, 'false') ->
+    PortReqId = wh_json:get_value(<<"port_request_id">>, DataJObj),
+    Attachments = wh_doc:attachments(wh_json:get_value(<<"port_request">>, DataJObj)),
+    AttachmentNames = wh_doc:attachment_names(Attachments),
```

Comment:
I don't think this will work. `get_value(<<"port_request">>, DataJObj)` would be the `{"_id":port_request",...,"_attachments":{...}}` doc, making Attachments the JSON object under the `_attachments` key. `wh_doc:attachment_names/1` wants to load attachments then get the names, which will fail here because `_attachments` won't exist in `Attachments`, so you'll only ever get `[]` as a return from `attachment_names/1`.